File: Leetcode/1768 Merge Strings Alternately.ts
function mergeAlternately(word1: string, word2: string): string {
    let merged = "";
    const maxLen = Math.max(word1.length, word2.length);
    for (let i = 0, j = 0; i < maxLen; i++) {
        if (i < word1.length) {
            merged += word1[i];
        } 
        if (i < word2.length) {
            merged += word2[i];
        }
    }
    return merged;
}

File: Leetcode/1768 Merge Strings Alternately.cs
public class Solution
{
    public string MergeAlternately(string word1, string word2)
    {
        var merged = new StringBuilder();
        for (int i = 0, j = 0; i < word1.Length || j < word2.Length;)
        {
            if (i < word1.Length)
            {
                merged.Append(word1[i++]);
            }
            if (j < word2.Length)
            {
                merged.Append(word2[j++]);
            }
        }
        return merged.ToString();
    }
}
